openssl/src/x509/mod.rs

```diff
@@ -629,6 +629,34 @@ impl X509Req {
 impl X509ReqRef {
     to_pem!(ffi::PEM_write_bio_X509_REQ);
     to_der!(ffi::i2d_X509_REQ);
+
+    pub fn version(&self) -> i32
+    {
+        unsafe {
+            compat::X509_REQ_get_version(self.as_ptr()) as i32
+        }
+    }
+
+    pub fn set_version(&mut self, value: i32) -> Result<(), ErrorStack>
```

Comment:
`X509_REQ` is reference counted, so we unfortunately can't mutate it. These are covered in a separate PR: https://github.com/sfackler/rust-openssl/pull/515/files#diff-3c34fc20efa0aafbeb82758e99f718e1R719